Makefile

```diff
@@ -517,18 +517,13 @@ dist/argosay:
 	mkdir -p dist
 	cp test/e2e/images/argosay/v2/argosay dist/
 
-.PHONY: kit
-kit:
-ifeq ($(shell command -v kit),)
-ifeq ($(shell uname),Darwin)
-	brew tap kitproj/kit --custom-remote https://github.com/kitproj/kit
-	brew install kit
-else
-	curl -q https://raw.githubusercontent.com/kitproj/kit/main/install.sh | tag=v0.1.8 sh
-endif
+.PHONY: task
+task:
+# update this in Nix when upgrading it here
+ifneq ($(USE_NIX), true)
+	npm i -g @go-task/cli@3.30.1
```

Comment:
Needs updating in the Nix file, happy to do that myself if you want?
Technically the Nix setup doesn't need task though because it already has something similar.

Reply:
@isubasinghe could you elaborate on this? You mentioned `devenv` before, but I didn't know that had task runner support (again, my Nix knowledge is pretty dated).
We discussed it at the Contributor meeting too if it could be used as a replacement on all environments, but no one knew the answer so we were looking to you for more info

Reply:
Checked it myself and it looks like [`devenv up`](https://devenv.sh/processes/) is a replacement for [`goreman`](https://github.com/mattn/goreman), but neither are full task runners, they just run processes concurrently (no `deps`, `sources`, `env`, etc).

`task` and `kit` are `make` replacements rather than `goreman` replacements.

Reply:
Yeah it does have a task runner, infact Nix can replace the Makefile and task/goreman/etc. 
https://github.com/argoproj/argo-workflows/blob/master/dev/nix/flake.nix#L393-L403 is how the processes to run are declared. 

There is a manual step of setting up the k8s cluster needed at the moment but even this can be fully integrated into Nix. 

That controller should have a version attached to it, I did not set this version. So production builds using Nix are not really possible yet. This is still very fixable though, more than happy to talk about it if this is something we want to do.

Reply:
> Yeah it does have a task runner, infact Nix can replace the Makefile and task/goreman/etc.
> https://github.com/argoproj/argo-workflows/blob/master/dev/nix/flake.nix#L393-L403 is how the processes to run are declared.

Yea I've seen that -- that's equivalent to `goreman` but that doesn't have `deps`, `sources`, `env` etc -- features of `task`, `kit`, and `make`. Unless that feature exists somewhere too and just isn't being used?